zebra-rpc/src/methods/get_block_template_rpcs/types/submit_block.rs

```diff
@@ -0,0 +1,49 @@
+//! Parameter and response types for the `submitblock` RPC.
+
+// Allow doc links to these imports.
+#[allow(unused_imports)]
+use crate::methods::get_block_template_rpcs::GetBlockTemplateRpc;
+
+/// Optional argument `jsonparametersobject` for `submitblock` RPC request
+///
+/// See notes for [`GetBlockTemplateRpc::submit_block`] method
+#[derive(Debug, serde::Deserialize)]
+pub struct JsonParameters {
+    pub(crate) _work_id: Option<String>,
+}
+
+/// Response to a `submitblock` RPC request.
+///
+/// Zebra never returns "duplicate-invalid", because it does not store invalid blocks.
+#[derive(Debug, PartialEq, Eq, serde::Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ErrorResponse {
+    /// Block was already committed to the non-finalized or finalized state
+    Duplicate,
+    /// Block was already added to the state queue or channel, but not yet committed to the non-finalized state
+    DuplicateInconclusive,
+    /// Block was already committed to the non-finalized state, but not on the best chain
+    Inconclusive,
+    /// Block rejected as invalid
+    Rejected,
+}
+
+/// Response to a `submitblock` RPC request.
+///
+/// Zebra never returns "duplicate-invalid", because it does not store invalid blocks.
+#[derive(Debug, PartialEq, Eq, serde::Serialize)]
+#[serde(untagged)]
+pub enum Response {
+    /// Block was not successfully submitted, return error
+    ErrorResponse(ErrorResponse),
+    /// Block successfully submitted, returns null
+    Accepted,
+}
+
+impl From<ErrorResponse> for Response {
+    fn from(error_response: ErrorResponse) -> Self {
+        Self::ErrorResponse(error_response)
+    }
+}
+#[derive(Debug, PartialEq, Eq, serde::Deserialize)]
+pub struct HexData(#[serde(with = "hex")] pub Vec<u8>);
```

Comment:
This seems like a type that could be used in multiple RPCs, can it go in its own module?

Also, can you open a ticket to refactor existing RPCs using `HexData`, and whatever error constructor we decide on?

Reply:
That's a really good point, yes. I moved it to its own module in [`cf4e937`](https://github.com/ZcashFoundation/zebra/pull/5526/commits/cf4e9376e4af691f2dc1183f53b1c95cdb91bcd5) and created issue #5548.